### PR TITLE
Complete all api documented in v2

### PR DESCRIPTION
**Pull request by i-am-unknown-81514525**
Do not push this to main until fastapi completed in v2. To test it before push, please use a disposible branch and remove after test complete

### CODE REVIEW

**Review by FloncDev — APPROVED** [review]
Would be ideal if someone more familiar with the API could review this to make sure it's all alright. Other than that, no comments.

**Comment by iauu** [comment]
I will look to see if I can improve some docstring meanwhile

File: bot/src/wrapper/_communication.py [hunk 1]
```diff
@@ -208,3 +211,60 @@ async def caller(session: aiohttp.ClientSession) -> ShopBuyOutput:
...
+                raise UnknownNetworkError(message)
+
+        return await make_request(session, caller)
+
+    @staticmethod
+    async def create_user(session: aiohttp.ClientSession, user_id: int) -> None:
+        """Create the user by user id through a direct HTTP request."""
+
+        async def caller(session: aiohttp.ClientSession) -> None:
+            async with session.get(f"user/{user_id}") as resp:
```
Comment (bensgilbert) [inline bot/src/wrapper/_communication.py]: Creating a user requires a POST request

File: bot/src/wrapper/_communication.py [hunk 2]
```diff
@@ -208,3 +211,60 @@ async def caller(session: aiohttp.ClientSession) -> ShopBuyOutput:
...
+        return await make_request(session, caller)
+
+    @staticmethod
+    async def update_user_experience(
+        session: aiohttp.ClientSession, user_id: int, src: UserIdExperiencePatchInput
+    ) -> UserIdExperiencePatchOutput:
+        """Get the user experience by user id through a direct HTTP request."""
+
+        async def caller(session: aiohttp.ClientSession) -> UserIdExperiencePatchOutput:
+            async with session.patch(f"user/{user_id}/experience", json=src) as resp:
```
Comment (bensgilbert) [inline bot/src/wrapper/_communication.py]: Technically the v2 doc states a POST request however PATCH makes more sense so my bad :)
I'll update the v3 doc

**Review by bensgilbert — CHANGES_REQUESTED** [review]

File: bot/src/wrapper/_communication.py (hunk 1 above)
Comment (iauu) [reply bot/src/wrapper/_communication.py]: Mb, forgot to change it when copy and paste

File: bot/src/wrapper/_communication.py (hunk 2 above)
Comment (iauu) [reply bot/src/wrapper/_communication.py]: It would be in a separate branch that base on `wrapperv2` call `wrapperv3`

Comment (iauu) [reply bot/src/wrapper/_communication.py]: Oh, oops, just notice I also implemented this wrong

File: bot/src/wrapper/_communication.py [hunk 3]
```diff
@@ -208,3 +211,60 @@ async def caller(session: aiohttp.ClientSession) -> ShopBuyOutput:
...
+                raise UnknownNetworkError(message)
+
+        return await make_request(session, caller)
+
+    @staticmethod
+    async def create_user(session: aiohttp.ClientSession, user_id: int) -> None:
+        """Create the user by user id through a direct HTTP request."""
+
+        async def caller(session: aiohttp.ClientSession) -> None:
+            async with session.post(f"user/{user_id}") as resp:
```
Comment (MilaDog) [inline bot/src/wrapper/_communication.py:240]: Technically, `user/{user_id}`, when looking at it, says `get user with this user_id`. Thus, making a post request to `user/{user_id}` to create a new user is not ideal. The endpoint itself, with a quick glance at it, show tell the reader everything they need to know about its function.

So, a POST to `/user` is good, even `/user/create`.

File: bot/src/wrapper/_communication.py [hunk 4]
```diff
@@ -208,3 +211,60 @@ async def caller(session: aiohttp.ClientSession) -> ShopBuyOutput:
...
+        return await make_request(session, caller)
+
+    @staticmethod
+    async def update_user_experience(
+        session: aiohttp.ClientSession, user_id: int, src: UserIdExperiencePatchInput
+    ) -> UserIdExperiencePatchOutput:
+        """Update the user experience by user id through a direct HTTP request."""
+
+        async def caller(session: aiohttp.ClientSession) -> UserIdExperiencePatchOutput:
+            async with session.post(f"user/{user_id}/experience", json=src) as resp:
```
Comment (MilaDog) [inline bot/src/wrapper/_communication.py:258]: PATCH is ideal over POST.

**Review by MilaDog — CHANGES_REQUESTED** [review]
Looks good. Gave two little comments, but it is more of a best practice.

File: bot/src/wrapper/_communication.py (hunk 4 above)
Comment (iauu) [reply bot/src/wrapper/_communication.py:258]: That was to documentation from V2, there are nothing I could change about, but it's different in V3 now

File: bot/src/wrapper/_communication.py (hunk 3 above)
Comment (iauu) [reply bot/src/wrapper/_communication.py:240]: Not quite sure what you meant by this, you mean the API? That was also from v2

Comment (MilaDog) [reply bot/src/wrapper/_communication.py:240]: Would be the API endpoint. Forgot this is the wrapper. I have not seen anything for the `user` endpoints, but my statement would apply more to when those endpoints are available.

But in summary, creating a user with POST `/user` is the ideal method.

Comment (iauu) [reply bot/src/wrapper/_communication.py:240]: Alright, maybe you could change the reference in v3 with the schema and params? I am not quite sure of the correct practice of this but I do know it is not too ideal.

**Review by MilaDog — APPROVED** [review]

**Review by bensgilbert — APPROVED** [review]

**Pull request merged**